contrib/win32/win32compat/shell-host.c

```diff
@@ -980,6 +1025,20 @@ ConsoleEventProc(HWINEVENTHOOK hWinEventHook,
 }
 
 DWORD 
+GetCurrentTerminalType(void)
+{
```

Comment:
1) Our VT Parser (part of win32-openssh code) gets kickin only when the console doesn't have the capability to ENABLE_VIRTUAL_TERMINAL_PROCESSING. From windows 10 onwards, the cmd.exe has the xterm capabilities which means we can set the TERM to xterm. 
I was testing on my windows 10 machine with TERM set to xterm, the scrolling doesn't happen after 46 lines with your current code changes.. 

2) At the server side, we shouldn't distinguish between the client (win32-openssh, putty, etc)..